test/README.md

```diff
@@ -114,17 +114,33 @@ starting with "test_ssl_":
 
 Run only test group 10:
 
-    $ make TESTS='10'
+    $ make TESTS='10' test
 
 Run all tests except the slow group (group 99):
 
-    $ make TESTS='-99'
+    $ make TESTS='-99' test
 
 Run all tests in test groups 80 to 99 except for tests in group 90:
 
-    $ make TESTS='[89]? -90'
+    $ make TESTS='[89]? -90' test
 
 To stochastically verify that the algorithm that produces uniformly distributed
 random numbers is operating correctly (with a false positive rate of 0.01%):
 
     $ ./util/wrap.sh test/bntest -stochastic
+
+Running Tests in Parallel
+-------------------------
+
+By default the test harness will execute the selected tests sequentially. When
+desirable it is possible to run test jobs in parallel via the `HARNESS_JOBS`
```

Comment:
Instead of "When desirable" I'd say: "For speeding up test execution"

Reply:
Well, it is debatable that this speeds up things. Running the jobs in parallel when you don't have enough cores or resources for it can actually slow it down!

Reply:
Why else would one desire to run (part of) the tests in parallel?

Reply:
At least in my experience with building OpenSSL (even when using ccache) using a dual-core CPU with hyperthreading (such that it virtually can run 4 parallel threads) is that `make -j4` is *much* faster than `make`.

Reply:
Yeah that is the goal, the problem I see with "For speeding up test execution it is possible to run test jobs in parallel" is that an unexperienced reader might wrongly conclude that enabling parallelization of tests will always deliver a speedup when the opposite can be true depending on the platform characteristics!

Reply:
(I also have the exact experience, and the tests are usually even less resource hungry than the compilation steps. Problem is if someone tries to enable this to speed up their testing on a single core ARM CPU or a container/vm restrained to a single core, attempting to multiplex might make things worse!

Reply:
I now see your point. Then how about this?

When desirable it is possible to run test jobs in parallel, which may speed up test execution depending on the platform characteristics, ...

Reply:
Fixup commit coming soon to address the feedback!

Reply:
@DDvO I went for this:

~~~diff
diff --git a/test/README.md b/test/README.md
index d046765494..b5080a3a6b 100644
--- a/test/README.md
+++ b/test/README.md
@@ -132,9 +132,12 @@ random numbers is operating correctly (with a false positive rate of 0.01%):
Running Tests in Parallel
-------------------------

By default the test harness will execute the selected tests sequentially.
{+Depending on the platform characteristics, running more than one test job in+}
{+parallel may speed up test execution.+} When desirable it is possible to run test
jobs in parallel via the `HARNESS_JOBS` environment [-variable.-]{+variable: its value should+}
{+be set to the number (a positive integer equal or greater than 1) of maximum+}
{+test jobs to run in parallel.+}

Depending on the Perl version different strategies could be adopted to select
which test recipes can be run in parallel.  In recent versions of Perl, unless
~~~

thoughts?

Reply:
I propose further adaptations to:

By default the test harness will execute the selected tests sequentially.
Depending on the platform characteristics, running test jobs in parallel may speed up test execution.
This can be requested by setting the `HARNESS_JOBS` environment variable to a positive integer value.
This specifies the maximum number of test jobs to run in parallel.

Reply:
Addressed in 57ea094fdc31440ccd956962abdd6388cf7f2c05